# Methods to remove NaNs
Call these methods to remove/replace NaNs AFTER all new features are created

In [1]:
import pickle

filename = 'flight_2019'

infile = open(filename,'rb')
flight_2019 = pickle.load(infile)
infile.close()

#### New dataframe with a smaller sample of 500,000 records

In [2]:
flight_sample = flight_2019.sample(n=1000000, random_state=1)

In [3]:
flight_sample.shape

(1000000, 38)

In [4]:
flight_sample.isnull().mean() * 100

mkt_unique_carrier      0.0000
branded_code_share      0.0000
mkt_carrier_fl_num      0.0000
op_unique_carrier       0.0000
tail_num                0.3525
op_carrier_fl_num       0.0000
origin_airport_id       0.0000
dest_airport_id         0.0000
crs_dep_time            0.0000
dep_time                1.8297
dep_delay               1.8300
taxi_out                1.8870
wheels_off              1.8870
wheels_on               1.9423
taxi_in                 1.9423
crs_arr_time            0.0000
arr_time                1.9423
arr_delay               2.1530
cancelled               0.0000
cancellation_code      98.0975
diverted                0.0000
dup                     0.0000
crs_elapsed_time        0.0000
actual_elapsed_time     2.1530
air_time                2.1530
flights                 0.0000
distance                0.0000
carrier_delay          81.1232
weather_delay          81.1232
nas_delay              81.1232
security_delay         81.1232
late_aircraft_delay    81.1232
first_de

## Fill in the tail_num NaNs

In [17]:
def tailNumNaN(flights_df):
    flights_df_cp = flights_df.copy()
    
    return flight_sample['tail_num'].fillna('0000')

In [10]:
flight_sample['tail_num'].describe()
flight_sample[flight_sample['tail_num'].isna()][['tail_num','cancelled']]
flight_sample[flight_sample['tail_num'] == '0000'][['tail_num','cancelled']]

Empty DataFrame
Columns: [tail_num, cancelled]
Index: []

In [14]:
# Set tail_num to 0000 for cancelled flights
flight_sample['tail_num'] = flight_sample['tail_num'].fillna('0000')

In [16]:
flight_sample[flight_sample['tail_num'] == '0000'][['tail_num','cancelled']]

Empty DataFrame
Columns: [tail_num, cancelled]
Index: []

* mkt_unique_carrier      0.0000
* branded_code_share      0.0000
* mkt_carrier_fl_num      0.0000
* op_unique_carrier       0.0000
* tail_num                0.3525
* op_carrier_fl_num       0.0000
* origin_airport_id       0.0000
* dest_airport_id         0.0000
* crs_dep_time            0.0000
dep_time                1.8297
dep_delay               1.8300
taxi_out                1.8870
wheels_off              1.8870
wheels_on               1.9423
taxi_in                 1.9423
* crs_arr_time            0.0000
arr_time                1.9423
arr_delay               2.1530
cancelled               0.0000
cancellation_code      98.0975
diverted                0.0000
* dup                     0.0000
* crs_elapsed_time        0.0000
actual_elapsed_time     2.1530
air_time                2.1530
* flights                 0.0000
* distance                0.0000
carrier_delay          81.1232
weather_delay          81.1232
nas_delay              81.1232
security_delay         81.1232
late_aircraft_delay    81.1232
first_dep_time         99.2994
total_add_gtime        99.2994
longest_add_gtime      99.2994
* fl_date                 0.0000


DROP dep_time                1.8297
DROP dep_delay               1.8300
DROP taxi_out                1.8870
DROP wheels_off              1.8870
DROP wheels_on               1.9423
DROP taxi_in                 1.9423
DROP arr_time                1.9423
DROP arr_delay               2.1530
DROP cancelled               0.0000
DROP cancellation_code      98.0975
DROP diverted                0.0000
DROP actual_elapsed_time     2.1530
DROP air_time                2.1530
DROP carrier_delay          81.1232
DROP weather_delay          81.1232
DROP nas_delay              81.1232
DROP security_delay         81.1232
DROP late_aircraft_delay    81.1232
DROP first_dep_time         99.2994
DROP total_add_gtime        99.2994
DROP longest_add_gtime      99.2994
